src/EventSourcingTests/Projections/Flattened/WriteTableWithGuidIdentifierProjection.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using Marten.Events.Projections.Flattened;
+
+namespace EventSourcingTests.Projections.Flattened;
+
+public class WriteTableWithGuidIdentifierProjection: FlatTableProjection
+{
+    public WriteTableWithGuidIdentifierProjection(): base("values", SchemaNameSource.DocumentSchema)
+    {
+        // Gotta have a primary key
+        Table.AddColumn<Guid>("id").AsPrimaryKey();
+
+        Project<ValuesSet>(cmd =>
+        {
+            // Derive column name by default, add column if does not exist
+            cmd.Map(x => x.A); // returns the column
```

Comment:
**QUESTION:** Wouldn't it be better to return here `StatementMap` to enable fluent chaining?  My understanding of the current `ColumnExpression` capabilities is to modify the schema, which I guess someone should not be doing here.

Reply:
Not in this case IMO. The call to `Map()` actually returns the Weasel column expression to do whatever you want to fine tune that particular column. You *could* go with a fluent interface instead, and make any customizations to the column be done through an optional `Action<ColumnExpression>` lambda. I think I'd vote to leave things the way it is, but don't have a very strong opinion if you think the fluent interface approach is really better. My preference with FI's these days is to keep the chaining very short.